data-service/templates/data_service.yml

```diff
@@ -0,0 +1,55 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Resources:
+
+  DynamoDBTable:
+    Type: AWS::DynamoDB::Table
+    Properties:
+     #TODO: change ddb table names to have region prefix passed in as a resource
+     TableName: DaemonEnvironments
+     AttributeDefinitions:
+       - AttributeName: environmentId
+         AttributeType: S
+       - AttributeName: environmentVersion
+         AttributeType: S
+       - AttributeName: cluster
+         AttributeType: S
+     KeySchema:
+       - AttributeName: "environmentId"
+         KeyType: HASH
+       - AttributeName: "environmentVersion"
+         KeyType: RANGE
+     GlobalSecondaryIndexes:
+       -
+         IndexName: "environment-cluster-index"
+         KeySchema:
+           -
+             AttributeName: "cluster"
+             KeyType: "HASH"
+         Projection:
+           ProjectionType: "KEYS_ONLY"
+         ProvisionedThroughput:
+           ReadCapacityUnits: 15
+           WriteCapacityUnits: 15
+     ProvisionedThroughput:
+       ReadCapacityUnits: 15
+       WriteCapacityUnits: 15
+
+  DynamoDBTable:
+    Type: AWS::DynamoDB::Table
+    Properties:
+     #TODO: change ddb table names to have region prefix passed in as a resource
+     TableName: DaemonEnvironmentsLatestVersion
```

Comment:
I wonder if "Latest" is the right term here. We need to keep track of the "target" version, which may be much smaller than the "Latest" version, if there have been no deployments, but there has been many updates.

In fact, we'd probably want to keep multiple version pointers for each environment, such as "Latest" (last update), "Target" (what we're currently enforcing), and "Previous" (for quick rollbacks).

Nit: I find the "Environment"/"EnvironmentLatestVersion" naming confusing. Conceptually the resource is an "Environment" which *has-many* "Versions", so calling the top level resource "Environment" and the individual versions "EnvironmentVersion" seems more clear to me.

Reply:
I'm not sure I follow your naming strategy either.